File: backend/revenant/src/main/java/com/santyman/revenant/dtos/EnemyResponse.java
package com.santyman.revenant.dtos;

public record EnemyResponse(
    Long id,
    Long id_map,
    Integer healthPoints,
    Integer damagePoints,
    Integer armorPoints,
    Integer goldReward,
    Integer xpReward,
    Integer speedAttackPoints,
    String name,
    String description
) {
    
}

File: frontend/revenant-app/src/game/interfaces/EnemyResponse.ts
/**
 * DTO representing an enemy returned by the backend.
 * Endpoint: GET /api/world/maps/enemies/{mapId}
 */
export interface EnemyResponse {
  id: number;
  id_map: number;
  healthPoints: number;
  damagePoints: number;
  armorPoints: number;
  goldReward: number;
  xpReward: number;
  speedAttackPoints: number;
  name: string;
  description: string;
}
